mm2src/common/mm_number.rs

```diff
@@ -32,11 +72,24 @@ impl<'de> Deserialize<'de> for MmNumber {
         enum MmNumberHelper {
             BigDecimal(BigDecimal),
             BigRational(BigRational),
+            Fraction(Fraction),
         }
 
         let ratio = match Deserialize::deserialize(deserializer)? {
             MmNumberHelper::BigDecimal(x) => from_dec_to_ratio(x),
-            MmNumberHelper::BigRational(x) => x
+            MmNumberHelper::BigRational(x) => x,
+            MmNumberHelper::Fraction(fraction) => {
+                let numer = BigInt::from_str(&fraction.numer).map_err(|e| {
```

Comment:
What do you think if the Fraction structure would have custom Serialization implementation with the numer and denom values validation?

Reply:
Makes sense, Fraction is public, so someone may use this type out of MmNumber context and it will miss the validation then. I can actually wrap BigInt in the type that will serde from/to String and use it as fields for Fraction. This way it will be guaranteed by type system that Fraction has numeric fields, will look better, thanks a lot!

Reply:
Done, please review again 🙂